modules/core/src/convert.cpp

```diff
@@ -4585,7 +4585,7 @@ cvtScaleHalf_<float, short>( const float* src, size_t sstep, short* dst, size_t
                     __m128i v_dst = _mm_cvtps_ph(v_src, 0);
 
                     _mm_storel_epi64((__m128i *)(dst + x), v_dst);
-#elif defined __GNUC__ && (defined __arm__ || defined __aarch64__)
+#elif defined __GNUC__ && (defined CV_NEON || defined __aarch64__)
```

Comment:
In Android NDK r9 llvm toolchans has float16x4_t and vcvt_f32_f16 in headers, but mentioned bits in flags are fine.
```
$ grep -R vcvt_f32_f16 /opt/android/android-ndk-r9/
/opt/android/android-ndk-r9/toolchains/llvm-3.2/prebuilt/linux-x86_64/lib/clang/3.2/include/arm_neon.h:__ai float32x4_t vcvt_f32_f16(float16x4_t __a) {
/opt/android/android-ndk-r9/toolchains/llvm-3.2/prebuilt/linux-x86_64/lib/clang/3.2/include/arm_neon.h:  return (float32x4_t)__builtin_neon_vcvt_f32_f16((int8x8_t)__a, 6); }
Binary file /opt/android/android-ndk-r9/toolchains/llvm-3.3/prebuilt/linux-x86_64/bin/clang++ matches
Binary file /opt/android/android-ndk-r9/toolchains/llvm-3.3/prebuilt/linux-x86_64/bin/clang matches
/opt/android/android-ndk-r9/toolchains/llvm-3.3/prebuilt/linux-x86_64/lib/clang/3.3/include/arm_neon.h:__ai float32x4_t vcvt_f32_f16(float16x4_t __a) {
/opt/android/android-ndk-r9/toolchains/llvm-3.3/prebuilt/linux-x86_64/lib/clang/3.3/include/arm_neon.h:  return (float32x4_t)__builtin_neon_vcvt_f32_f16((int8x8_t)__a, 6); }
```

I will try to use CMake "try_compile" here.